contracts/cw721-base/src/lib.rs

```diff
@@ -25,133 +16,93 @@ pub use cw_ownable::{Action, Ownership, OwnershipError};
+        state::{DefaultOptionCollectionInfoExtension, DefaultOptionMetadataExtension},
+    };
 
     // This makes a conscious choice on the various generics used by the contract
     #[cfg_attr(not(feature = "library"), entry_point)]
     pub fn instantiate(
         deps: DepsMut,
         env: Env,
         info: MessageInfo,
-        msg: InstantiateMsg,
-    ) -> Result<Response, ContractError> {
-        cw2::set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
-
-        let tract = Cw721Contract::<Extension, Empty, Empty, Empty>::default();
-        tract.instantiate(deps, env, info, msg)
+        msg: Cw721InstantiateMsg<DefaultOptionCollectionInfoExtension>,
+    ) -> Result<Response, Cw721ContractError> {
+        let contract = Cw721Contract::<
+            DefaultOptionMetadataExtension,
+            Empty,
+            Empty,
+            Empty,
+            DefaultOptionCollectionInfoExtension,
+        >::default();
+        contract.instantiate(deps, env, info, msg, CONTRACT_NAME, CONTRACT_VERSION)
     }
 
     #[cfg_attr(not(feature = "library"), entry_point)]
     pub fn execute(
         deps: DepsMut,
         env: Env,
         info: MessageInfo,
-        msg: ExecuteMsg<Extension, Empty>,
-    ) -> Result<Response, ContractError> {
-        let tract = Cw721Contract::<Extension, Empty, Empty, Empty>::default();
-        tract.execute(deps, env, info, msg)
+        msg: Cw721ExecuteMsg<
+            DefaultOptionMetadataExtension,
+            Empty,
+            DefaultOptionCollectionInfoExtension,
+        >,
+    ) -> Result<Response, Cw721ContractError> {
+        let contract = Cw721Contract::<
+            DefaultOptionMetadataExtension,
+            Empty,
+            Empty,
+            Empty,
+            DefaultOptionCollectionInfoExtension,
+        >::default();
+        contract.execute(deps, env, info, msg)
     }
 
     #[cfg_attr(not(feature = "library"), entry_point)]
-    pub fn query(deps: Deps, env: Env, msg: QueryMsg<Empty>) -> StdResult<Binary> {
-        let tract = Cw721Contract::<Extension, Empty, Empty, Empty>::default();
-        tract.query(deps, env, msg)
+    pub fn query(
+        deps: Deps,
+        env: Env,
+        msg: Cw721QueryMsg<Empty, DefaultOptionCollectionInfoExtension>,
+    ) -> StdResult<Binary> {
+        let contract = Cw721Contract::<
+            DefaultOptionMetadataExtension,
+            Empty,
+            Empty,
+            Empty,
+            DefaultOptionCollectionInfoExtension,
+        >::default();
+        contract.query(deps, env, msg)
     }
 
     #[cfg_attr(not(feature = "library"), entry_point)]
-    pub fn migrate(deps: DepsMut, _env: Env, _msg: Empty) -> Result<Response, ContractError> {
-        // make sure the correct contract is being upgraded, and it's being
-        // upgraded from the correct version.
-        cw2::assert_contract_version(deps.as_ref().storage, CONTRACT_NAME, EXPECTED_FROM_VERSION)?;
```

Comment:
Side note here: old migration logic was very strict. Expected version is `v0.16.0`, so only contracts of this version was able to migrate!

Reply:
I think this was from when we implemented cw-ownable, only cw 0.16.0 had the valid migrate path

Reply:
> I think this was from when we implemented cw-ownable, only cw 0.16.0 had the valid migrate path to v.017.0. As in, I don't think you can go from an earlier version directly to v.0.17.0

With this version, you can migrate from any version to latest :sunglasses: